File: fibonacci.c
#include <stdio.h>
#include <time.h>

int fibonacci(int);



int main(void){

    clock_t t;

    t = clock();
    int n = fibonacci(40);
    t = clock() - t;
    double total_time = ((double)t)/CLOCKS_PER_SEC;

    printf("The %d number of fibonacci is %d\nIt took %f",40,n,total_time);
    return 0;
}

int fibonacci(int n){

    if(n <= 1){
        return n;
    }
    else{
        return fibonacci(n-1) + fibonacci(n-2);
    }
}

File: fibonacci.py
from time import time


n = 20
numCalc = 15


def fibonacci(num):
    if num <= 1:
        return num
    else:
        return fibonacci(n-1) + fibonacci(n-2)

def fibCrunch(i):
    initialTime = time()
    
    answer = fibonacci(n)

    endTime = time()

    totalTime = endTime - initialTime

    print("Iteration {} runtime: {}".format(i,totalTime))

    return totalTime


def main():
    

    result = 0

    mean = 0

    lab = raw_input("Please enter the lab: ")

    computer = raw_input("\nPlease enter the computer number: ")

    print("\nFibonacci {} = 1836311903\n".format(n))

    try:
        f = open("Lab_{}_Comp_{}.txt".format(lab,computer),"w")
        f.write("Lab: {}\tComputer Number: {}\n".format(lab,computer))
        for it in range(0,numCalc):
            result = fibCrunch(it)
            f.write("Runtime {}:\t{}\n".format(it+1,result))
            mean += result
        print("Mean:\t\t\t{} nanoseconds".format(mean/numCalc))
        f.write("Mean:\t\t{} nanoseconds\n".format(mean/numCalc))
    except IOError:
        print("An error occured trying to read the file.")
    


main()



